imports/plugins/included/discount-codes/server/methods/methods.js

```diff
@@ -193,6 +193,24 @@ export const methods = {
     //   enabled: true
     // };
 
+    // check to ensure discounts can only apply to single shop carts
+    // TODO: Remove this check after implementation of shop-by-shop discounts
+    const Collection = Reaction.Collections[collection];
+    const objectToApplyDiscount = Collection.findOne({ _id: id });
+    const items = objectToApplyDiscount && objectToApplyDiscount.items;
+    // loop through all items and filter down to unique shops (in order to get participating shops in the order/cart)
+    const uniqueShopObj = items.reduce((shopObj, item) => {
+      if (!shopObj[item.shopId]) {
+        shopObj[item.shopId] = true;
+      }
+      return shopObj;
+    }, {});
+    const participatingShops = Object.keys(uniqueShopObj);
+
+    if (participatingShops.length > 1) {
+      throw new Meteor.Error(400, "discounts.multiShopError", "Discounts cannot be applied to a multi-shop cart or order");
```

Comment:
We don't pass a number to `Meteor.Error` any more. I think we should probably pass `not-implemented` there as I have been using that whenever there is a marketplace feature that's not yet done.

Reply:
Cool. Adding that now